src/main/java/uk/ac/ebi/ena/readtools/loader/fastq/DataSpot.java

```diff
@@ -522,4 +525,19 @@
             .append( null == quals ? "null" : quals.length() )
             .toString();
     }
+
+    private void handleInvalidBases(String bases) {
+        Matcher matcher = p_bases.matcher("");
+
+        //remove the trailing '+' symbol.
+        bases = bases.substring(0, bases.length() - 1);
+
+        Set<Character> invalidBaseChars = bases.chars()
+                .mapToObj(intChar -> (char)intChar)
+                // '+' symbol is added because its present in the regex pattern.
+                .filter(base -> !matcher.reset(String.valueOf(base) + "+").matches())

```

Comment:
Ok, the '+' emerges here as well. Would it be possible to rewrite this so that '+' is not needed as part of the matching?

Reply:
its certainly possible but not probably not worth the trouble as it would require changes in more than places. especially, the line reading code which is complex enough as it is.